kore/app/exec/Main.hs

```diff
@@ -355,12 +362,24 @@ parseKoreExecOptions startTime =
         <*> optional parseKoreMergeOptions
         <*> optional parseRtsStatistics
         <*> parseBugReport
+
     SMT.Config { timeOut = defaultTimeOut } = SMT.defaultConfig
-    readSMTTimeOut = do
-        i <- auto
-        if i <= 0
-            then readerError "smt-timeout must be a positive integer."
-            else return $ SMT.TimeOut $ Limit i
+    SMT.Config { resetInterval = defaultResetInterval } = SMT.defaultConfig
+
+    readPositiveInteger ctor optionName = do
```

Comment:
Some of the functions here (including this one) are needed in `kore-repl` as well, so they appear duplicated there. Could we have a clean-up issue to move them all to a separate, common module (maybe `GlobalMain.hs`)? Or would that not work for some reason?

Reply:
We should just write a separate command-line option parser for `SMT.Config`. It can go in `Options.SMT` in the library. Please do this as part of this pull request.

I don't want to put new things in `GlobalMain`; actually, I want to move everything out of there.